src/processing/resources/index.js

```diff
@@ -37,8 +37,8 @@ export async function process (ctx, customCharset) {
             var urlReplacer = getResourceUrlReplacer(ctx);
             var processed   = processors[i].processResource(decoded, ctx, charset, urlReplacer);
 
-            if (processed === null)
-                return null;
+            if (processed && processed.err)
+                return await process(ctx, processed.rightCharset);
```

Comment:
\cc @churkin @AlexanderMoskovkin Maybe you will have better idea for this mechanism